Classes/Sync/Dropbox/DropboxTransferManager.swift

```diff
@@ -0,0 +1,257 @@
+//
+//  DropboxTransferManager.swift
+//  MobileOrg
+//
+//  Created by Mario Martelli on 16.12.16.
+//
+//  This program is free software; you can redistribute it and/or
+//  modify it under the terms of the GNU General Public License
+//  as published by the Free Software Foundation; either version 2
+//  of the License, or (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+//
+
+
+import Foundation
+import SwiftyDropbox
+
+@objc final class DropboxTransferManager: NSObject {
+
+  var activeTransfer:TransferContext? = nil
+  var transfers:[TransferContext] = []
+  var _active = false
+  var paused = false
+
+  var active:Bool{
+    get { return _active }
+    set {_active = newValue }
+  }
+  
+  func queueSize() -> Int {
+    return transfers.count
+  }
+
+  func busy() -> Bool {
+    return (transfers.count > 0 || active)
+  }
+
+  func abort() {
+    transfers.removeAll()
+  }
+
+  // Workaround to suffice to TransferManager protocol
+
+
+  static let instance = DropboxTransferManager()
+
+  override init() {
+    let filePath = Bundle.main.path(forResource: "AppKey", ofType: "plist")
+    let plist = NSDictionary(contentsOfFile:filePath!)
+    let dkist = plist?["Dropbox API Key"] as? NSDictionary
+    let appKey = dkist?.object(forKey: "AppKey") as! String
+    DropboxClientsManager.setupWithAppKey(appKey)
```

Comment:
When I run this on my phone it end up with following crush.

```
fatal error: unexpectedly found nil while unwrapping an Optional value
2016-12-18 11:52:43.491941 MobileOrg[2940:909018] fatal error: unexpectedly found nil while unwrapping an Optional value
```

The last piece of code before some hell is following line in `OAuth.swift` in `SwiftyDropbox`.

```
self.dauthRedirectURL = URL(string: "db-\(appKey)://1/connect")!
```